vtm/src/org/oscim/map/Animator2.java

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright 2013 Hannes Janetzek
+ * Copyright 2016 Stephan Leuschner
+ * Copyright 2016 devemux86
+ * Copyright 2016 Izumi Kawashima
+ * Copyright 2017 Wolfgang Schramm
+ * Copyright 2018 Gustl22
+ *
+ * This file is part of the OpenScienceMap project (http://www.opensciencemap.org).
+ *
+ * This program is free software: you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ * PARTICULAR PURPOSE. See the GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.oscim.map;
+
+import org.oscim.backend.CanvasAdapter;
+import org.oscim.core.Point;
+import org.oscim.core.Tile;
+import org.oscim.renderer.MapRenderer;
+import org.oscim.utils.animation.DragForce;
+import org.oscim.utils.animation.Easing;
+import org.oscim.utils.ThreadUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.oscim.utils.FastMath.clamp;
+
+public class Animator2 extends Animator{
+    static final Logger log = LoggerFactory.getLogger(Animator2.class);
+
+    public final static int ANIM_KINETIC = 1 << 5;
+
+    /**
+     * The minimum changes that are pleasant for users.
+     */
+    private static final float DEFAULT_MIN_VISIBLE_CHANGE_PIXELS = 0.5f;
+
+    private static final float FLING_FRICTION_MOVE = 0.9f;
+
+    private final Point mScrollRatio = new Point();
+    private final Point mMovePoint = new Point();
+    private final DragForce mFlingScrollForce = new DragForce();
+    private final DragForce mFlingScaleForce = new DragForce();
+    private final DragForce mFlingRotateForce = new DragForce();
+
+    private float mScrollDet2D = 1f;
+    private long mFrameStart = -1;
+
+    public Animator2(Map map) {
+        super(map);
+    }
+
+    @Override
+    public void animateFling(float velocityX, float velocityY,
+                             int xmin, int xmax, int ymin, int ymax) {
+        kineticScroll(velocityX, velocityY, xmin, xmax, ymin, ymax);
+    }
+
+    /**
+     * @param velocityX the x velocity depends on screen resolution
+     * @param velocityY the y velocity depends on screen resolution
+     */
+    public void kineticScroll(float velocityX, float velocityY,
+                                  int xmin, int xmax, int ymin, int ymax) {
```

Comment:
What's its difference with `Animator.animateFling`?
Cannot see any and some docs could help solve the confusion in posts. 🙂

Reply:
The difference is that it uses same sheme of the other ease animations (especially in `update`). The result keeps the same. Next time I'll do more docs then.